### PR TITLE
fix(android): Fix OSK rotation issues

### LINKED ISSUE #9191: bug(android): Keyboard position does not fit properly in the screen after changing to Landscape Orientation in Android 12.0 OS
### Describe the bug

I noticed that after changing the orientation to landscape mode the keyboard position changes and it does not fit properly in the keyboard frame.

### Reproduce the bug

1. Install Keyman 17.0.133-alpha build. 
2. Open Keyman In-App. 
3. We may notice that the Get Started dialog opens along with keyman application. 
4. Change the Orientation from Portrait to Landscape. 
5. Close the Get Started dialog. 

Here, I noticed that the Keyboard does not fit in its position. 

Note: It is only reproducible in Android 12.0 OS. 

I have attached the Screenshot as well as the Video file for reference. 

..in Android 12.0 Mobile

   
android12mobile

..in Android 12.0 Nexus 7 Tab

  
NexusAndroid

..Android 12.0 Mobile Video

https://github.com/keymanapp/keyman/assets/19683143/9dc16e68-6125-4943-960d-9a576608cdd6

..Android 12.0 Nexus Video

https://github.com/keymanapp/keyman/assets/19683143/c8c11169-f623-405b-97c8-1d3e4ef46810

### Expected behavior

After changing the orientation to landscape the keyboard position should fit into the screen. 

### Related issues

_No response_

### Keyman apps

- [X] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.133-alpha

### Operating system

Android 12.0

### Device

Android Mobile (Emulator), Nexus 7 (Emulator)

### Target application

_No response_

### Browser

_No response_

### Keyboard name

_No response_

### Keyboard version

_No response_

### Language name

_No response_

### Additional context

_No response_

### LINKED ISSUE #9978: bug(android): The keyboard disappeared after changing the orientation of the on-screen keyboard from Portrait to Landscape in Android 13
### Describe the bug

I observed that adjusting the height of the on-screen keyboard caused it to vanish in the Keyman In-App. This issue is specific to the most recent Android 13 version.

### Reproduce the bug

1. Install Keyman 17.0.208-alpha build.
2. Open Keyman In-App.
3. Open the Settings menu. 
4. Open Adjust keyboard height menu.
5. Set the height to the maximum size. 
6. Revert to Keyman Home screen. 
7. Change the Orientation from Portrait to Landscape. 

Here, I noticed that the On-Screen Keyboard disappeared from the keyman home screen. Rotating to Portrait Orientation does not show up OSK in the home screen. 

..Set to Maximum height

   

..OSK is missing after the rotation (P-to-L)

   

..OSK is still missing after the change to the default position (L-to-P)

   

### Expected behavior

The Keyboard should be visible after changing the orientation from Portrait-to-Landscape.

### Related issues

_No response_

### Keyman apps

- [X] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.208-alpha

### Operating system

Android 13

### Device

Samsung Galaxy A23 5G Mobile

### Target application

_No response_

### Browser

_No response_

### Keyboard name

English - EuroLatin (SIL) Keyboard

### Keyboard version

2.0.4

### Language name

English

### Additional context

_No response_

### LINKED ISSUE #10241: bug(android): swapping keyboards via picker in landscape results in poor kbd layout
From user-testing of #7324:

> * **TEST_REGRESSION_ANDROID_KBD (FAILED):** Retested with the attached PR build (Keyman 17.0.227-alpha-test-7324 in the Android Mobile device (ver 13) and here is my observation: 1. While testing  TEST_INAPP_LATIN_KEYBOARD_LANDSCAPE test in the lanscape mode the OSK shrinks it width after using the globe key. Here, What I did is :
> 1. In the Landscape mode, typed some letter using the OSK. 
> 2. Used Backspace and Enter button after typing each words. 
> 3. Clicked the globe key. 
> 4. Selected the same keyboard from the Picker menu. 
>
> Here, I noticed that the OSK shrinks its width on the Keyman home page. Seems to be an issue. 
> 
> 

_Originally posted by @bharanidharanj in https://github.com/keymanapp/keyman/issues/7324#issuecomment-1851940191_

I have reproduced this on `master` - thus it is not a `feature-gestures` issue.  There's a chance this is related to recent ✨ Android optimizations.

Of note:  simply tapping the globe-key to swap keyboards does _not_ cause the issue.  Using the picker menu causes it.

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #10241 
Fixes #9978 
Fixes #9191 

There were multiple bugs related to OSK not filling the full width or not appearing after rotating the device orientation.

The main two fixes on this PR:
* Disable multi-window (where modern Android devices try to split the screen so dimensions are "half"
* The existing device orientation code would sporadically report PORTRAIT when the device was LANDSCAPE. Changing to getRotation() seems to have more consistent behavior.

## User Testing
**Setup** - Install the PR build of Keyman for Android on the device/emulator. Some tests specify a specific Android SDK level

* **TEST_LANDSCAPE_KEYBOARD_PICKER** - Verifies menu to adjust keyboard height works
1. Install the PR build of Keyman for Android on an Android device/emulator
2. with the device in Landscape mode, Launch Keyman and dismiss the Get Started menu
3. In the Keyman app, type some letter using the OSK
4. Longpress the globe key to display the keyboard picker menu
5. Select the same keyboard from the picker menu
6. Verify OSK is still full width

* **TEST_OSK_ROTATION_ANDROID_12**
1. Install the PR build of Keyman for Android on an Android device/emulator Android 12 (SDK 32) 
2. Open Keyman In-App.
3. Open the Settings menu.
4. Open Adjust keyboard height menu.
5. Set the height to the maximum size.
6. Revert to Keyman Home screen.
7. Change the Orientation from Portrait to Landscape.
8. Verify OSK displays and is full width.
 
* **TEST_OSK_ROTATION_ANDROID_13**
1. Install the PR build of Keyman for Android on an Android device/emulator Android 13 (SDK 33) 
2. Open Keyman In-App.
3. Open the Settings menu.
4. Open Adjust keyboard height menu.
5. Set the height to the maximum size.
6. Revert to Keyman Home screen.
7. Change the Orientation from Portrait to Landscape.
8. Verify OSK displays and is full width.

### CODE REVIEW

File: android/KMAPro/kMAPro/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -71,6 +71,7 @@
       android:exported="true"
       android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|screenSize|smallestScreenSize"
       android:label="@string/app_name"
+      android:resizeableActivity="false"
```
Comment (darcywong00) [inline android/KMAPro/kMAPro/src/main/AndroidManifest.xml:74]: This disables the multi-window mode so Android won't use split-screen sizes.

**Comment by bharanidharanj** [comment]
* **TEST_OSK_ROTATION_ANDROID_13 (PASSED):** Tested with the attached PR build (Keyman 17.0243-alpha-test-10373) in the Android Mobile device (Android version 13 / Samsung Galaxy A23 5g) and here is my observation : Verified that the OSK displayed with full width in the landscape mode. Seems to be working fine.

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_LANDSCAPE_KEYBOARD_PICKER (PASSED):**  Tested with the attached PR build (Keyman 17.0243-alpha-test-10373) in the Android 5.0 / API 21 emulator and here is my observation: 1. Verified that the OSK displayed with full width in the landscape mode. Seems to be working fine.

**Review by jahorton — APPROVED** [review]

**Comment by jahorton** [comment]
#10054 tagged at least two of the issues being fixed by this PR.  Granted, I don't think we have a consistent repro for it... but how likely do you think this PR fixing #10054 as well would be?

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.245-alpha](https://keyman.com/downloads/releases/alpha/17.0.245)

**Pull request merged**